From c93bc94b651e2aa2dec7418e304c66870ac98106 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Mon, 5 Oct 2015 19:54:38 +0530
Subject: [PATCH] USBC-PD:Delaying the auto crc enablement on connect

Early auto crc enablement  results in port partner issuing
hard reset due to delay in extcon event processing in DPM.
Hence enable the auto crc after handling the connect event in DPM.

Change-Id: I0b2a9fd06ebecc5db8e9cdc9b7bdb9a9089b0195
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-6025
Reviewed-on: https://android.intel.com:443/419768
---
 drivers/usb/typec/pd/devpolicy_mgr.c | 4 ++++
 drivers/usb/typec/usb_typec_detect.c | 3 ---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/typec/pd/devpolicy_mgr.c b/drivers/usb/typec/pd/devpolicy_mgr.c
index aa31c1b..a3635fe 100644
--- a/drivers/usb/typec/pd/devpolicy_mgr.c
+++ b/drivers/usb/typec/pd/devpolicy_mgr.c
@@ -645,6 +645,10 @@ static void dpm_handle_ext_cable_event(struct devpolicy_mgr *dpm,
 			mutex_unlock(&dpm->role_lock);
 
 			dpm_notify_policy_evt(dpm, dpm_evt);
+			typec_enable_autocrc(dpm->phy,
+				dpm_evt == DEVMGR_EVENT_UFP_CONNECTED
+				|| dpm_evt == DEVMGR_EVENT_DFP_CONNECTED);
+
 		} else
 			mutex_unlock(&dpm->role_lock);
 
diff --git a/drivers/usb/typec/usb_typec_detect.c b/drivers/usb/typec/usb_typec_detect.c
index 8b62ce9..617164f 100644
--- a/drivers/usb/typec/usb_typec_detect.c
+++ b/drivers/usb/typec/usb_typec_detect.c
@@ -513,7 +513,6 @@ static void detect_dfp_work(struct work_struct *work)
 	if (detect_src_attached(&phy->cc1, &phy->cc2)) {
 		use_cc = get_active_cc(phy->cc1.rd, phy->cc2.rd);
 		typec_setup_cc(phy, use_cc, TYPEC_STATE_ATTACHED_DFP);
-		typec_enable_autocrc(detect->phy, true);
 		/* enable VBUS */
 	} else if (detect_audio_attached(&phy->cc1, &phy->cc2)) {
 		dev_info(detect->phy->dev, "Audio Accessory Detected");
@@ -606,8 +605,6 @@ static void detect_update_ufp_state(struct typec_detect *detect)
 	detect->state = DETECT_STATE_ATTACHED_UFP;
 	mutex_unlock(&detect->lock);
 
-	typec_enable_autocrc(detect->phy, true);
-
 	typec_detect_notify_extcon(detect,
 				TYPEC_CABLE_USB_SNK, true);
 	typec_detect_notify_extcon(detect,
-- 
1.9.1

